### ISSUE TITLE
IsConnected failing on iOS simulator

### ISSUE THREAD

**Issue by richard-edwards** [jamesmontemagno/ConnectivityPlugin#37]
I'm trying to do some testing in the iOS simulator (iPhone 5, 9.3) with the Network Link Conditioner on OSX El Capitan. The `CrossConnectivity.Current.IsConnected` is coming back as true even though I have the Network Link Conditioner set to 100% failure. Basically my app gets past the IsConnected test and tries to hit my endpoint and craps out which is what I'm trying to avoid.

Does it work on the simulators?

Anything I can do from my end to further diagnose where I might be going wrong? I've installed the plugin on all projects, iOS/Android/Shared and I'm accessing it from shared code.

Running Xam Studio 6.1.1

**Comment by jamesmontemagno** [comment]
Interesting, I have never tested it that way. The API may not actually be returning the correct API. 

I test it by actually turning off wifi on my mac to simulator. I actually test it on a device before releasing though.

**Comment by jamesmontemagno** [comment]
In the documentation it says:

If your app uses Reachability to detect network availability, you may experience some unexpected results while using the Network Link Conditioner. As such, any reachability behavior under Airplane mode or WWan / WiFi distinctions is something that should be tested separately from network conditioning.

Which is what this uses, so most likely it isn't compatible. Sorry.

**Comment by richard-edwards** [comment]
Hi James,

Thanks for the feedback. Just wanted to clarify your approach to simulating no network on the simulator. You said 'actually turning off wifi on my mac to simulator'. Do you mean you are turning off wifi on your mac totally and that drops connection for the simulator, or do you do it a different way.

**Comment by jamesmontemagno** [comment]
Yeah, so the simulator just uses the network on your mac itself. So if you cut off the network on your mac it will cut off the network on your simulator. I usually test on a real device though with airplane mode.

**Comment by richard-edwards** [comment]
So I tested it the way you are doing and it's working as expected. Thanks again.

I did find a few OSX commands to turn it off instead of using the gui:

`networksetup -setairportpower airport off`

`networksetup -setairportpower airport on`

Also, I tried the Network Link Conditioner again, this time just using to limit bandwidth and test for bad connections, low latency etc. because the app I'm working on is being used in Brazil and that seems to work great.

Thanks again for all the work you're doing with Xamarin. The plugins are all very helpful, keep up the great work.
